### PR TITLE
Delete PVCs on scale down.

### PR DESCRIPTION
**Pull request by jhamacher-indeed**

### CODE REVIEW

File: src/test/java/com/indeed/operators/rabbitmq/reconciliation/TestRabbitMQClusterReconciler.java [hunk 1]
```diff
@@ -99,4 +106,55 @@ public void shouldSkipReonciliationNullResource() throws InterruptedException {
...
+        final Reconciliation rec = new Reconciliation(NAME, NAME, NAMESPACE, "type");
+
+        final StatefulSet originalStatefulSet = new StatefulSet(
+                "apps/v1",
+                "StatefulSet",
+                new ObjectMetaBuilder().build(),
+                new StatefulSetSpecBuilder().withReplicas(4).build(),
+                null
+        );
+        final StatefulSet scaledStatefulSet = new StatefulSet(
```
Comment (jhamacher-indeed) [inline src/test/java/com/indeed/operators/rabbitmq/reconciliation/TestRabbitMQClusterReconciler.java]: This is never used, I forgot to remove it.

File: src/main/java/com/indeed/operators/rabbitmq/reconciliation/RabbitMQClusterReconciler.java [hunk 2]
```diff
@@ -19,7 +17,6 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.IOException;
```
Comment (jhamacher-indeed) [inline src/main/java/com/indeed/operators/rabbitmq/reconciliation/RabbitMQClusterReconciler.java]: Just cleaning up unused imports.

File: src/test/java/com/indeed/operators/rabbitmq/reconciliation/TestRabbitMQClusterReconciler.java [hunk 3]
```diff
@@ -11,24 +11,33 @@
...
+import static com.indeed.operators.rabbitmq.Constants.RABBITMQ_STORAGE_NAME;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyZeroInteractions;
 import static org.mockito.Mockito.when;
 
 @ExtendWith(MockitoExtension.class)
 public class TestRabbitMQClusterReconciler {
 
+    private final String NAME = "name";
+    private final String NAMESPACE = "namespace";
```
Comment (jhamacher-indeed) [inline src/test/java/com/indeed/operators/rabbitmq/reconciliation/TestRabbitMQClusterReconciler.java:41]: Currently only used in one method but they'll be used more widely in #18 .

File: src/test/java/com/indeed/operators/rabbitmq/reconciliation/TestRabbitMQClusterReconciler.java [hunk 4]
```diff
@@ -56,12 +65,12 @@
     private RabbitMQClusterReconciler reconciler;
 
     @BeforeEach
-    public void setup() {
```
Comment (jhamacher-indeed) [inline src/test/java/com/indeed/operators/rabbitmq/reconciliation/TestRabbitMQClusterReconciler.java:61]: I'm a big believer in minimum access and technically these don't need to be public.

File: src/test/java/com/indeed/operators/rabbitmq/reconciliation/TestRabbitMQClusterReconciler.java [hunk 5]
```diff
@@ -56,12 +65,12 @@
...
+    void setup() {
         reconciler = new RabbitMQClusterReconciler(clusterFactory, controller, secretsController, servicesController, statefulSetController, podDisruptionBudgetController, persistentVolumeClaimController, shovelReconciler);
     }
 
     @Test
-    public void shouldSkipReonciliation() throws InterruptedException {
+    void shouldSkipReconciliation() throws InterruptedException {
```
Comment (jhamacher-indeed) [inline src/test/java/com/indeed/operators/rabbitmq/reconciliation/TestRabbitMQClusterReconciler.java:78]: Can't believe I never noticed the typo before.

File: src/main/java/com/indeed/operators/rabbitmq/reconciliation/RabbitMQClusterReconciler.java [hunk 6]
```diff
@@ -67,9 +64,10 @@ public void reconcile(final Reconciliation reconciliation) throws InterruptedExc
 
         if (shouldReconcile(resource)) {
             final RabbitMQCluster cluster = clusterFactory.fromCustomResource(resource);
+            final int currentReplicaCount = determineCurrentReplicaCount(resource);
```
Comment (jhamacher-indeed) [inline src/main/java/com/indeed/operators/rabbitmq/reconciliation/RabbitMQClusterReconciler.java:71]: Really the key to this entire change - we need to save the replica count before reconciling the StatefulSet.

File: src/test/java/com/indeed/operators/rabbitmq/reconciliation/TestRabbitMQClusterReconciler.java (hunk 4 above)
Comment (bfink13) [reply src/test/java/com/indeed/operators/rabbitmq/reconciliation/TestRabbitMQClusterReconciler.java:61]: yes very good

File: src/test/java/com/indeed/operators/rabbitmq/reconciliation/TestRabbitMQClusterReconciler.java [hunk 7]
```diff
@@ -99,4 +106,48 @@ public void shouldSkipReonciliationNullResource() throws InterruptedException {
...
+                )
+        );
+
+        // This call will happen twice.  In both cases it will occur before the StatefulSet has been patched, hence it
+        // will reflect the origin unscaled replica count.
+        when(statefulSetController.get(NAME, NAMESPACE)).thenReturn(originalStatefulSet);
+
+        reconciler.reconcile(rec);
+
+        verify(persistentVolumeClaimController).delete(RABBITMQ_STORAGE_NAME + "-" + NAME + "-3", NAMESPACE);
```
Comment (bfink13) [inline src/test/java/com/indeed/operators/rabbitmq/reconciliation/TestRabbitMQClusterReconciler.java:156]: i'd add a 
```
verifyNoMoreInteractions(persistentVolumeClaimController);
```
to verify that only the one volume is deleted

**Review by bfink13 — APPROVED** [review]

**Pull request merged**